Bug Report: jamfpro_macos_configuration_profile_plist_generator Provider Crash

## Summary

When running `terraform plan` with the `jamfpro_macos_configuration_profile_plist_generator` resource, the provider crashes with the following error:

```
panic: interface conversion: interface {} is nil, not bool
...
The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).PlanResourceChange call.
```

## Description

A clear and concise description of the bug:
- The jamfpro provider crashes during `terraform plan` when using a valid configuration for the `jamfpro_macos_configuration_profile_plist_generator` resource.
- All required and optional fields are set and not null.
- Only one `payload_content` block is used.
- The error persists with the latest provider version (v0.33.0).

## Steps to Reproduce

1. Clone or set up a Terraform project with the provided configuration (see below).
2. Run `terraform init` to initialize the workspace.
3. Run `terraform plan`.
4. Observe the provider crash and error output.

## Expected Behavior

Terraform should successfully generate a plan and not crash. The provider should handle the resource configuration gracefully.

## Actual Behavior

Terraform fails to generate a plan. The provider crashes with a panic and does not respond to the PlanResourceChange call.

## Environment

- **OS**: macOS
- **Terraform Version**: (please fill in, e.g., 1.5.0)
- **Provider Version**: deploymenttheory/jamfpro v0.33.0
- **Other dependencies**: None

## Additional Context

- Downgrading the provider is not possible (older versions unavailable).
- The error occurs with the latest provider version.
- All fields are set and not null.

## Possible Solution

The error appears to be a bug in the provider's handling of the resource schema, possibly related to a nil value or type assertion in the Go code. Please review the provider's implementation for handling optional or missing fields in the resource.

## Screenshots or Videos

N/A (see logs below)

## Logs & Console Output

```
panic: interface conversion: interface {} is nil, not bool
...
The plugin encountered an error, and failed to respond to the plugin6.(*GRPCProvider).PlanResourceChange call.
```

## Related Issues

N/A

## Priority

- High (Causes significant problems, needs prompt attention)

---

**Sample Configuration**

See the attached summary in `BUG_REPORT_SUMMARY.md` for full configuration and variable definitions.

[BUG_REPORT_SUMMARY.md](https://github.com/user-attachments/files/25340692/BUG_REPORT_SUMMARY.md)

--- a/internal/common/plist/conversion.go
+++ b/internal/common/plist/conversion.go
@@ -71,21 +71,21 @@ func mapSchemaToProfile(d *schema.ResourceData) *ConfigurationProfile {
 			PayloadVersion:      val["payload_version"].(int),
 		}
 
-		settings := val["setting"].([]any)
-		if len(settings) == 0 {
-			return out
+		settings, ok := val["setting"].([]any)
+		if !ok || len(settings) == 0 {
+			out.PayloadContent = append(out.PayloadContent, payloadContentStruct)
+			continue
 		}
 
 		payloadContentStruct.ConfigurationItems = make(map[string]any, 0)
 		for _, s := range settings {
 			settingMap := s.(map[string]any)
 			dictionary := parseNestedDictionary(settingMap["dictionary"])
-			payloadContent := map[string]any{
-				"key":        settingMap["key"].(string),
-				"value":      GetTypedValue(settingMap["value"]),
-				"dictionary": dictionary,
+			if dictionary != nil && len(dictionary) > 0 {
+				payloadContentStruct.ConfigurationItems[settingMap["key"].(string)] = dictionary
+			} else {
+				payloadContentStruct.ConfigurationItems[settingMap["key"].(string)] = GetTypedValue(settingMap["value"])
 			}
-			payloadContentStruct.ConfigurationItems[settingMap["key"].(string)] = payloadContent
 		}
 
 		out.PayloadContent = append(out.PayloadContent, payloadContentStruct)
@@ -103,8 +103,14 @@ func parseNestedDictionary(dict any) map[string]any {
 	result := make(map[string]any)
 	dictionary := dict.([]any)
 	for _, item := range dictionary {
-		entry := item.(map[string]any)
-		key := entry["key"].(string)
+		entry, ok := item.(map[string]any)
+		if !ok {
+			continue
+		}
+		key, ok := entry["key"].(string)
+		if !ok {
+			continue
+		}
 		value := GetTypedValue(entry["value"])
 		if nestedDict, ok := entry["dictionary"].([]any); ok {
 			value = parseNestedDictionary(nestedDict)

--- a/internal/services/macos_configuration_profile_plist_generator/resource_custom_diff.go
+++ b/internal/services/macos_configuration_profile_plist_generator/resource_custom_diff.go
@@ -12,7 +12,7 @@ import (
 
 // mainCustomDiffFunc orchestrates all custom diff validations for macOS config profiles.
 func mainCustomDiffFunc(ctx context.Context, diff *schema.ResourceDiff, i any) error {
-	if diff.Get("payload_validate").(bool) {
+	if v, ok := diff.GetOk("payload_validate"); ok && v.(bool) {
 		if err := validatePayloadIdentifers(ctx, diff, i); err != nil {
 			return err
 		}
@@ -50,7 +50,11 @@ func mainCustomDiffFunc(ctx context.Context, diff *schema.ResourceDiff, i any) e
 }
 
 func normalizePayloadState(_ context.Context, diff *schema.ResourceDiff, _ any) error {
-	diff.SetNew("payloads", plist.NormalizePayloadState(diff.Get("payloads").(string)))
+	payloads, ok := diff.GetOk("payloads")
+	if !ok {
+		return nil
+	}
+	diff.SetNew("payloads", plist.NormalizePayloadState(payloads.(string)))
 	return nil
 }
 
@@ -156,7 +160,10 @@ func validateAllComputersScope(_ context.Context, diff *schema.ResourceDiff, _ a
 	}
 
 	scope := scopeRaw.([]any)[0].(map[string]any)
-	allComputers := scope["all_computers"].(bool)
+	allComputers, ok := scope["all_computers"].(bool)
+	if !ok {
+		return nil
+	}
 
 	if allComputers {
 		fieldsToCheck := []string{"computer_ids", "computer_group_ids"}
@@ -180,7 +187,10 @@ func validateAllUsersScope(_ context.Context, diff *schema.ResourceDiff, _ any) 
 	}
 
 	scope := scopeRaw.([]any)[0].(map[string]any)
-	allJssUsers := scope["all_jss_users"].(bool)
+	allJssUsers, ok := scope["all_jss_users"].(bool)
+	if !ok {
+		return nil
+	}
 
 	if allJssUsers {
 		fieldsToCheck := []string{"jss_user_ids", "jss_user_group_ids", "building_ids", "department_ids"}

--- a/internal/services/macos_configuration_profile_plist_generator/resource_state.go
+++ b/internal/services/macos_configuration_profile_plist_generator/resource_state.go
@@ -35,8 +35,10 @@ func updateState(d *schema.ResourceData, resp *jamfpro.ResourceMacOSConfiguratio
 
 	d.Set("site_id", resp.General.Site.ID)
 
-	profile := plist.NormalizePayloadState(resp.General.Payloads)
-	if err := d.Set("payloads", profile); err != nil {
+	payloadsHCL, err := plist.ConvertPlistToHCL(resp.General.Payloads)
+	if err != nil {
+		diags = append(diags, diag.FromErr(err)...)
+	} else if err := d.Set("payloads", payloadsHCL); err != nil {
 		diags = append(diags, diag.FromErr(err)...)
 	}
 

